cpp/src/reductions/struct_minmax_util.cuh

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/detail/structs/utilities.hpp>
+#include <cudf/detail/utilities/vector_factories.hpp>
+#include <cudf/table/row_operators.cuh>
+#include <cudf/table/table_device_view.cuh>
+
+namespace cudf {
+namespace reduction {
+namespace detail {
+
+/**
+ * @brief Binary operator ArgMin/ArgMax with index values into the input table.
+ */
+struct row_arg_minmax_fn {
+  size_type const num_rows;
+  row_lexicographic_comparator<nullate::DYNAMIC> const comp;
+  bool const arg_min;
+
+  row_arg_minmax_fn(size_type const num_rows,
+                    table_device_view const& table,
```

Comment:
Is `num_rows` here the same as `table.num_rows()`.  `table_device_view::num_rows()` is callable on the host, so you could drop the parameter.

Reply:
Done.